phpBB/phpbb/style/manager.php

```diff
@@ -0,0 +1,441 @@
+<?php
+/**
+*
+* This file is part of the phpBB Forum Software package.
+*
+* @copyright (c) phpBB Limited <https://www.phpbb.com>
+* @license GNU General Public License, version 2 (GPL-2.0)
+*
+* For full copyright and license information, please see
+* the docs/CREDITS.txt file.
+*
+*/
+
+namespace phpbb\style;
+
+use phpbb\style\exception;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+* The style manager provide means to manage styles
+*/
+class manager
+{
+	protected $cache;
+
+	/** @var ContainerInterface */
+	protected $container;
+
+	protected $db;
+	protected $config;
+	protected $filesystem;
+	protected $styles_table;
+	protected $users_table;
+	protected $phpbb_root_path;
+
+	/** @var \phpbb\textformatter\cache_interface */
+	protected $text_formatter_cache;
+
+	protected $reserved_style_names = array('adm', 'admin', 'all');
+
+	protected $styles_path;
+	protected $styles_path_absolute = 'styles';
+
+	public function __construct(\phpbb\cache\service $cache = null, \phpbb\config\config $config, ContainerInterface $container, \phpbb\db\driver\driver_interface $db, \phpbb\filesystem\filesystem_interface $filesystem, $styles_table, $users_table, $phpbb_root_path)
+	{
+		$this->cache = $cache;
+		$this->config = $config;
+		$this->container = $container;
```

Comment:
I will leave it there for now, because it is only used to get text_formatter.cache, and probably it will need to be moved to acp and cli instead of manager